yellowbrick/features/histogram.py

```diff
@@ -0,0 +1,191 @@
+        bin: number of bins the user wants, default=4
+        """
+        # get the number of samples
+        samples, = X.shape
+
+        # get the index for retrieving values for the bins
+        index = int(round(samples/self.bins))
+
+        # sort the samples from low to high
+        sorted_X = np.sort(X)
+
+        # a list to store the values for the binning reference
+        values = []
+
+        # get the values for the binning reference
+        for i in range(0, (self.bins-1)):
+            value = sorted_X[index * (i + 1)]
+            values.append(value)
+
+        return values
+
+    def draw(self, X, **kwargs):
+        """
+        Draws a histogram with the reference value for binning as vetical lines
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x 1
+            A matrix of n instances with 1 feature
+        """
+
+        # draw the histogram
+        self.ax.hist(X, bins=50, color="#6897bb", **kwargs)
+
+        # add vetical line with binning reference values
+        self.ax.vlines(self.get_vline_value(X), 0, 1, transform=self.ax.get_xaxis_transform(), colors='r')
+        print("The binning reference values are:", self.get_vline_value(X))
+
+    def fit(self, X, **kwargs):
+        """
+        Sets up X for the histogram and checks to
+        ensure that X is of the correct data type
+        Fit calls draw
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x 1
+            A matrix of n instances with 1 feature
+        kwargs: dict
+            keyword arguments passed to Scikit-Learn API.
+        """
+
+        #throw an error if X has more than 1 column
+        if is_dataframe(X):
+            nrows, ncols = X.shape
+
+            if ncols > 1:
+                raise YellowbrickValueError((
+                    "X needs to be an ndarray or DataFrame with one feature, "
+                    "please select one feature from the DataFrame"
+                ))
+
+        # Handle the feature name if it is None.
+        if self.feature is None:
+
+            # If X is a data frame, get the columns off it.
+            if is_dataframe(X):
+                self.feature = X.columns
+
+            else:
+                self.feature = ['x']
+
+        self.draw(X)
+        return self
+
+
+    def poof(self, **kwargs):
+        """
+        Creates the labels for the feature and target variables
+        """
+
+        self.ax.set_xlabel(self.feature)
+        self.ax.set_ylabel(self.target)
+        self.finalize(**kwargs)
+
+    def finalize(self, **kwargs):
+        """
+        Finalize executes any subclass-specific axes finalization steps.
+        The user calls poof and poof calls finalize.
+        Parameters
+        ----------
+        kwargs: generic keyword arguments.
+        """
+
+        plt.setp(self.ax.get_xticklabels(), visible=True)
+        plt.setp(self.ax.get_yticklabels(), visible=True)
```

Comment:
Because visualizers may be working on a specific axes object in a multi-axes figure, I usually prefer to contain all drawing to `self.ax` -- in this case I believe it's:

```python
for tk in self.ax.get_xticklabels():
    tk.set_visible(True) 
```

I know that's a bit more work, but it makes the visualizer much more robust and safer to use in a notebook environment.